### ISSUE TITLE
Is there anything wrong with training?

### ISSUE THREAD

**Issue by garyliu0816** [edgarschnfld/CADA-VAE-PyTorch#2]
when using the cmd to train the model:
>python single_experiment.py --dataset CUB --num_shots 0 --generalized True

It works fine before 10 epoch, but then the loss increase from 3358 to 14401.
>epoch 10 | iter 0          | loss 3358.
>epoch 99 | iter 100      | loss 14401.

Is this situation right？

**Comment by edgarschnfld** [comment]
This situation is right. The reason is, that the _weight_ of all loss terms is continuously increased (but at different rates and over different intervals). For example, the weight of the cross-reconstruction loss is more than doubled from epoch 20 to 70.

**Comment by garyliu0816** [comment]
Thank you for your answer. 
If the loss continuously increase, how can I determine when to stop training?
> This situation is right. The reason is, that the _weight_ of all loss terms is continuously increased (but at different rates and over different intervals). For example, the weight of the cross-reconstruction loss is more than doubled from epoch 20 to 70.

**Comment by edgarschnfld** [comment]
You can do that via trial and error on the validation set. I have experienced that the VAE-training is very robust to overfitting, and that the performance does not get worse with longer training times. For example, it makes a difference if you train 40 or 100 epochs (the performance will increase), but there is not much difference in performance between 100 and 1000 epochs (overall performance is basically stable). In effect, it means that trying out some different number of epochs should suffice to know when to stop training.

**Comment by garyliu0816** [comment]
Thanks again, this work is awesome.
